src/types/entities/ui.ts

```diff
@@ -206,8 +206,18 @@ export default class Ui {
 		}
 	}
 
-	updatePlayer(line: number, text: string, when: number) {
-		this._boxes[BOXES.ALL_PLAYERS].setLine(line, Ui.genText(text, when))
+	insertTopCallback(box: BOXES): (text: string, when?: number) => void {
+		return (text: string, when?: number) => {
+			this._boxes[box].insertTop(Ui.genText(text, when))
+			// render the changes
+			this._screen.render()
+		}
+	}
+
+	updatePlayer(line: number, text: string, when: number | undefined) {
```

Comment:
When is undefined for pre-populated overlays from the specified highlight overlays, in case you were wondering.   They show up in All Players without a timestamp until they are seen to actually participated.

Reply:
Refer #5